llvm/utils/TableGen/SubtargetEmitter.cpp

```diff
@@ -902,8 +902,15 @@ SubtargetEmitter::FindWriteResources(const CodeGenSchedRW &SchedWrite,
   for (Record *WR : ProcModel.WriteResDefs) {
     if (!WR->isSubClassOf("WriteRes"))
       continue;
-    if (AliasDef == WR->getValueAsDef("WriteType") ||
-        SchedWrite.TheDef == WR->getValueAsDef("WriteType")) {
+    // If there is no AliasDef and we find a match, we can early exit since
+    // there is no need to verify whether there are resources defined for both
+    // SchedWrite and its alias.
+    Record *WRDef = WR->getValueAsDef("WriteType");
+    if (!AliasDef && SchedWrite.TheDef == WRDef) {
```

Comment:
What if AliasDef is non-null. We won't update ResDef if SchedWrite.TheDef == WRDef.

```suggestion
Record *WRDef = WR->getValueAsDef("WriteType");
if (AliasDef == WRDef || SchedWrite.TheDef == WRDef) {
  if (ResDef)
    PrintFatalError...
   ResDef = WR;
   // If there is no AliasDef and we find a match, we can early exit...
   if (!AliasDef)
     break;
```